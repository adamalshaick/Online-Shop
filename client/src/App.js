import React, { Component } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import jwt_decode from "jwt-decode";

import { Provider } from "react-redux";
import store from "./store";

import Landing from "./components/layout/Landing";
import Register from "./components/auth/Register";

class App extends Component {
  render() {
    return (
      <Provider store={store}>
        <Router>
          <>
            <Route exact path="/" component={Landing} />
            <Route exact path="/register" component={Register} />
          </>
        </Router>
      </Provider>
    );
  }
}

export default App;
