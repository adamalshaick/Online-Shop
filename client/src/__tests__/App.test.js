import React from "react";
import { shallow } from "enzyme";
import App from "../App";
import toJson from "enzyme-to-json";

const wrapper = shallow(<App />);

describe("render component", () => {
  it("renders component", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
