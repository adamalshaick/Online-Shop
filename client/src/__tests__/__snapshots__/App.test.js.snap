// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render component renders component 1`] = `
<BrowserRouter>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/register"
    />
    <Route
      component={[Function]}
      exact={true}
      path="/login"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/items"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/profile/:handle"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/dashboard"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/create-profile"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/sell-item"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/cart"
    />
    <Connect(PrivateRoute)
      component={[Function]}
      exact={true}
      path="/edit-profile"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</BrowserRouter>
`;
